### PR TITLE
#Issue SB-7025 feat:View all button functionality (Phase I)

### PR DESCRIPTION
**Pull request by ArchanaKumari01**

### CODE REVIEW

File: src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export class DataDrivenFilterComponent implements OnInit, OnDestroy, OnChanges {
   @Input() ignoreQuery = [];
   @Input() showSearchedParam = true;
   @Input() enrichFilters: object;
+  @Input() cacheExists =  false;
```
Comment (anoophm) [inline src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts]: change name of input, viewAllMode

File: src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts [hunk 2]
```diff
@@ -255,12 +256,21 @@ export class DataDrivenFilterComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   resetFilters() {
+    let queryParams = {};
     if (!_.isEmpty(this.ignoreQuery)) {
       this.formInputData = _.pick(this.formInputData, this.ignoreQuery);
+      queryParams = this.formInputData;
+    } else if (this.cacheExists) {
+      const data = this._cacheService.get('searchQuery');
+      _.forOwn(data.request.filters, (queryValue, queryParam) => {
```
Comment (anoophm) [inline src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts]: queryParam to queryKey

File: src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts [hunk 3]
```diff
@@ -255,12 +256,21 @@ export class DataDrivenFilterComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   resetFilters() {
+    let queryParams = {};
```
Comment (anoophm) [inline src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts]: no need of this

File: src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts [hunk 4]
```diff
@@ -255,12 +256,21 @@ export class DataDrivenFilterComponent implements OnInit, OnDestroy, OnChanges {
...
     if (!_.isEmpty(this.ignoreQuery)) {
       this.formInputData = _.pick(this.formInputData, this.ignoreQuery);
+      queryParams = this.formInputData;
+    } else if (this.cacheExists) {
```
Comment (anoophm) [inline src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts]: separate out this else if case

File: src/app/client/src/app/modules/core/components/search/search.component.ts [hunk 5]
```diff
@@ -82,6 +82,7 @@ export class SearchComponent implements OnInit {
    */
   onEnter(key) {
     this.key = key;
+    this.queryParam = {};
```
Comment (anoophm) [inline src/app/client/src/app/modules/core/components/search/search.component.ts]: confirm with namritha on this

File: src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts [hunk 6]
```diff
@@ -309,8 +310,20 @@ export class LearnPageComponent implements OnInit, OnDestroy {
     this.playerService.playContent(event.data.metaData);
   }
   viewAll(event) {
-    const sectionUrl = 'learn/view-all/' + event.name;
-    this.router.navigate([sectionUrl, 1], {});
+    this.cacheService.set('searchQuery', JSON.parse(event.searchQuery), {
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts]: viewAllQuery

Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts]: format query then store

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 7]
```diff
@@ -1,15 +1,287 @@
...
+    this.toasterService = toasterService;
+    this.config = config;
+    this.coursesService = coursesService;
+    this.route.onSameUrlNavigation = 'reload';
+    this.sortingOptions = this.config.dropDownConfig.FILTER.RESOURCES.sortingOptions;
+  }
 
   ngOnInit() {
+    this.filterType = this.config.appConfig.course.filterType;
+    this.pageLimit = this.config.appConfig.SEARCH.PAGE_LIMIT;
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: this.config.appConfig.viewAll.PAGE_LIMIT;

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 8]
```diff
@@ -1,15 +1,287 @@
...
+    combineLatest(this.activatedRoute.params, this.activatedRoute.queryParams).pipe(
+      map(results => ({ params: results[0], queryParams: this.manipulateData(results[1])})),
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: this.manipulateQueryParam(results[1])

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 9]
```diff
@@ -1,15 +1,287 @@
...
+   */
+  private searchService: SearchService;
+  /**
+  * To call resource service which helps to use language constant
+  */
+  private resourceService: ResourceService;
+  /**
+   * To get url, app configs
+   */
+  public config: ConfigService;
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: public configService: ConfigService;

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 10]
```diff
@@ -1,15 +1,287 @@
...
+      pageid: this.activatedRoute.snapshot.data.telemetry.pageid,
+    };
+    this.sortIntractEdata = {
+      id: 'sort',
+      type: 'click',
+      pageid: this.activatedRoute.snapshot.data.telemetry.pageid,
+    };
+  }
+  private manipulateData(results) {
+    this.cache = this._cacheService.exists('searchQuery');
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: remove

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 11]
```diff
@@ -1,15 +1,287 @@
...
+      _.forOwn(results, (queryValue, queryParam) => {
+        this.filters[queryParam] = queryValue;
+      });
+      this.filters = _.omit(this.queryParams, ['key', 'sort_by', 'sortType', 'defaultSortBy']);
+    }
+    return this.queryParams;
+  }
+
+  private getContentList(request) {
+    const route = this.route.url.split('/view-all');
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: do this in ngoninit

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 12]
```diff
@@ -1,15 +1,287 @@
...
+    this.closeUrl = '/' + route[0].toString();
+    const name = route[1].split('/');
+    this.sectionName = name[1];
+    this.pageNumber = request.params.pageNumber;
+    const requestParams = {
+      filters: _.pickBy(this.filters, value => value.length > 0),
+      limit: this.pageLimit,
+      pageNumber: request.params.pageNumber,
+      query: request.queryParams.key,
+      softConstraints: { badgeAssertions: 1 },
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: not needed

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 13]
```diff
@@ -1,15 +1,287 @@
...
+    const route = this.route.url.split('/view-all');
+    this.closeUrl = '/' + route[0].toString();
+    const name = route[1].split('/');
+    this.sectionName = name[1];
+    this.pageNumber = request.params.pageNumber;
+    const requestParams = {
+      filters: _.pickBy(this.filters, value => value.length > 0),
+      limit: this.pageLimit,
+      pageNumber: request.params.pageNumber,
+      query: request.queryParams.key,
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: not needed

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 14]
```diff
@@ -1,15 +1,287 @@
...
+      softConstraints: { badgeAssertions: 1 },
+      sort_by: _.get(request.queryParams, 'sortType') ?
+        { [request.queryParams.sort_by]: request.queryParams.sortType } : this.defaultSortBy
+    };
+    return combineLatest(
+      this.searchService.contentSearch(requestParams),
+      this.coursesService.enrolledCourseData$,
+      (contentData, enrolledCourseData) => ({ contentData, enrolledCourseData })
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: use rxjs 6 syntax

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 15]
```diff
@@ -1,15 +1,287 @@
...
+    const name = route[1].split('/');
+    this.sectionName = name[1];
+    this.pageNumber = request.params.pageNumber;
+    const requestParams = {
+      filters: _.pickBy(this.filters, value => value.length > 0),
+      limit: this.pageLimit,
+      pageNumber: request.params.pageNumber,
+      query: request.queryParams.key,
+      softConstraints: { badgeAssertions: 1 },
+      sort_by: _.get(request.queryParams, 'sortType') ?
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: use request.queryParams. sortType

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 16]
```diff
@@ -1,15 +1,287 @@
...
+      map(results => ({ params: results[0], queryParams: this.manipulateData(results[1])})),
+      mergeMap((data) => {
+        this.setTelemetryImpressionData();
+        this.setInteractEventData();
+        return this.getContentList(data);
+      })
+    ).subscribe((response: any) => {
+      this.showLoader = false;
+      if (response.contentData.result.count && response.contentData.result.content) {
+        this.showLoader = false;
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: not needed

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 17]
```diff
@@ -1,15 +1,287 @@
...
+        this.noResult = false;
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: not needed

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 18]
```diff
@@ -1,15 +1,287 @@
...
+        this.totalCount = response.contentData.result.count;
+        this.pager = this.paginationService.getPager(response.contentData.result.count, this.pageNumber, this.pageLimit);
+        this.searchList = this.formatData(response);
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: formatSearchresults

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 19]
```diff
@@ -1,15 +1,287 @@
...
+        this.setInteractEventData();
+        return this.getContentList(data);
+      })
+    ).subscribe((response: any) => {
+      this.showLoader = false;
+      if (response.contentData.result.count && response.contentData.result.content) {
+        this.showLoader = false;
+        this.noResult = false;
+        this.totalCount = response.contentData.result.count;
+        this.pager = this.paginationService.getPager(response.contentData.result.count, this.pageNumber, this.pageLimit);
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: why we need this

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 20]
```diff
@@ -1,15 +1,287 @@
...
+    });
   }
 
+  setTelemetryImpressionData() {
+    this.telemetryImpression = {
+      context: {
+        env: this.activatedRoute.snapshot.data.telemetry.env
+      },
+      edata: {
+        type: this.activatedRoute.snapshot.data.telemetry.type,
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: use _.get()

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 21]
```diff
@@ -1,15 +1,287 @@
...
+    };
+    this.sortIntractEdata = {
+      id: 'sort',
+      type: 'click',
+      pageid: this.activatedRoute.snapshot.data.telemetry.pageid,
+    };
+  }
+  private manipulateData(results) {
+    this.cache = this._cacheService.exists('searchQuery');
+    this.queryParams = results;
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: do this in subscribe

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 22]
```diff
@@ -1,15 +1,287 @@
...
+    this.filters = {};
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: do this while creating

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 23]
```diff
@@ -1,15 +1,287 @@
...
+   */
+  pager: IPagination;
+  /**
+   *url value
+   */
+  queryParams: any;
+  /**
+ *search filters
+ */
+  filters: any;
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: filters ={};

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 24]
```diff
@@ -1,15 +1,287 @@
...
+  loaderMessage: ILoaderMessage;
+  /**
+   * Contains returned object of the pagination service
+   * which is needed to show the pagination on inbox view
+   */
+  pager: IPagination;
+  /**
+   *url value
+   */
+  queryParams: any;
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: queryParams = {};

File: src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts [hunk 25]
```diff
@@ -1,15 +1,287 @@
...
   ngOnInit() {
+    this.filterType = this.config.appConfig.course.filterType;
+    this.pageLimit = this.config.appConfig.SEARCH.PAGE_LIMIT;
+    combineLatest(this.activatedRoute.params, this.activatedRoute.queryParams).pipe(
+      map(results => ({ params: results[0], queryParams: this.manipulateData(results[1])})),
+      mergeMap((data) => {
+        this.setTelemetryImpressionData();
+        this.setInteractEventData();
+        return this.getContentList(data);
+      })
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-feature/components/view-all/view-all.component.ts]: takeUntil

File: src/app/client/src/app/modules/core/components/search/search.component.ts (hunk 5 above)
Comment (ArchanaKumari01) [reply src/app/client/src/app/modules/core/components/search/search.component.ts]: confirmed with namrita no filters should be carried to search page.

**Pull request merged**